apps/organizations/models.py

```diff
@@ -907,74 +854,96 @@ def mailchimp_integration_connected(self):
 
     def ensure_mailchimp_store(self) -> None:
         if not self.mailchimp_store:
-            logger.info("Creating store for rp_id=[%s]", self.id)
-            self.make_mailchimp_store()
-        else:
-            logger.info("Store already exists for rp_id=[%s]", self.id)
-
-    def ensure_mailchimp_one_time_contribution_product(self) -> None:
-        if not self.mailchimp_one_time_contribution_product:
-            logger.info("RP with ID %s does not have a one-time contributor product. Attempting to create", self.id)
-            self.make_mailchimp_one_time_contribution_product()
-        else:
-            logger.info("One-time contribution product already exists for rp_id=[%s]", self.id)
-
-    def ensure_mailchimp_recurring_contribution_product(self) -> None:
-        if not self.mailchimp_recurring_contribution_product:
-            logger.info("RP with ID %s does not have a recurring contributor product. Attempting to create", self.id)
-            self.make_mailchimp_recurring_contribution_product()
+            logger.info("Creating store for RP %s", self.id)
+            RevenueProgramMailchimpClient(self).create_store()
         else:
-            logger.info("Recurring contribution product already exists for rp_id=[%s]", self.id)
-
-    def ensure_mailchimp_contributor_segment(self) -> None:
-        # TODO: [DEV-3579] Handle edge case where segment has been defined in Mailchimp but the ID is not saved in the RP
-        if not self.mailchimp_contributor_segment:
-            logger.info(
-                "Creating %s segment for rp_id=[%s]",
-                self.mailchimp_contributor_segment_name,
-                self.id,
+            logger.info("Store already exists for RP %s", self.id)
+
+    def ensure_mailchimp_contribution_product(self, type: Literal["one_time", "recurring"]) -> None:
+        if not getattr(self, f"mailchimp_{type}_contribution_product"):
```

Comment:
Need to define default otherwise will cause error if no value is there.

```suggestion
        if not getattr(self, f"mailchimp_{type}_contribution_product", None):
```